src/helpers/token.ts

```diff
@@ -95,6 +95,23 @@ export function unwrapSOLInstruction(
   return null;
 }
 
+/**
+ * Get token account balance with fallback for newly created accounts
+ * @param connection - Solana connection
+ * @param tokenAccount - The token account address
+ * @param accountCreationIx - The instruction that creates the account (if any)
+ * @returns The token account balance as a string, or "0" if the account is being created
+ */
+export async function getTokenAccountBalance(
+  connection: Connection,
+  tokenAccount: PublicKey,
+  accountCreationIx?: TransactionInstruction
+): Promise<string> {
+  return accountCreationIx
```

Comment:
we don't do this. should be just get token balance action.
and should do another check outside this function. 
example:
`if(ataExitesed) getTokenBalance() else 0;`